### ISSUE TITLE
Do we need double quotes for input arguments in the SHELL scripts?

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/sct_tutorial_data#12]
Current example scrips have double quotes for CLI arguments:

https://github.com/spinalcordtoolbox/sct_tutorial_data/blob/dde7a4bfc18f539225af78a8a465ab191b9cff18/multi_subject/process_data.sh#L117-L123

However, I don't think it is required. I find double quotes removes clarity, and might give bad ideas to users who might use them in their terminal. Previously I've used scripts without quotes, and it worked fine ([example](https://github.com/spine-generic/spine-generic/blob/c8c8408715f101414ca08c6bcf29a5d5b87df90a/process_data.sh#L146-L157)).

**Comment by joshuacwnewton** [comment]
Double quotes are added from the recommendations of shellcheck:

```
Line 60:
    sct_label_vertebrae -i ${file}.nii.gz -s ${file_seg}.nii.gz -c t2 -qc "${PATH_QC}" -qc-subject "${SUBJECT}"
                           ^-- SC2086 (info): Double quote to prevent globbing and word splitting.
                                             ^-- SC2086 (info): Double quote to prevent globbing and word splitting.
```

The relevant rule is [SC2086](https://www.shellcheck.net/wiki/SC2086). Without double-quotes, the following will happen:

> The first argument [will be split] by IFS (Internal Field Separators, i.e. spaces, tabs and line feeds). Then, each of them [will be expanded] as if it was a glob. Lastly, all the resulting strings and filenames [will be joined] with spaces. 

In other words, if e.g. `${file_t2}` contains any spaces, tabs, line feeds, or globbing [wildcard characters](https://tldp.org/LDP/abs/html/globbingref.html) (`*`, `?`, `[`, `]`, `^`), then unexpected behavior may occur.

----

> Previously I've used scripts without quotes, and it worked fine ([example](https://github.com/spine-generic/spine-generic/blob/c8c8408715f101414ca08c6bcf29a5d5b87df90a/process_data.sh#L146-L157)).

However, as you point out, if we can be 100% sure that none of the special characters are present, then double-quotes aren't necessary.

Looking at this script in particular (`multi-subject/process_data.sh`), since we control the data being used, we already know that the exact paths and filenames of the various subjects do _not_ contain special characters. So, I think it would be okay to remove the double-spaces.

The only way I think this could cause issues is if:

- Later on, our users copy the example scripts into a folder that has a space (e.g. `My Documents`)
- Users have installed SCT in a location with a space in the path, and thus `$SCT_DIR` contains a space. See:
  https://github.com/spinalcordtoolbox/sct_tutorial_data/blob/dde7a4bfc18f539225af78a8a465ab191b9cff18/multi_subject/process_data.sh#L156-L157

But if we are okay with these risks, then I think we can remove the double-quotes.

**Comment by jcohenadad** [comment]
> Double quotes are added from the recommendations of shellcheck:

Right, I was suspecting something like this. The problem is that users copy/paste, from/to the terminal, edit the variables, replace them, add/remove some, etc. all these manipulations resulting in higher probability for "forgetting" a double quote somewhere, resulting in issues like this one:

<img width="1337" alt="image" src="https://github.com/spinalcordtoolbox/sct_tutorial_data/assets/2482071/ed7d8d1a-c713-4f44-a6bd-6cba2233ca3f">

Notice the "{PATH_QC}" folder, which was created because I forgot one double quote in the script after modifying it. 

I consider myself more experienced than the average user, so I anticipate a _lot_ of similar mistakes. 

That being said, I do see the value in covering issues with spaces in file names (plenty of our users have that, unfortunately). 

On the other hand, not silencing errors if there is a path with spaces is also a way to change users' behaviour in the long run. 

Maybe we should discuss this during our next SCT meeting

**Comment by joshuacwnewton** [comment]
This was addressed by #13 but never linked. Closing.
